libs/core/tests/unit_tests/runnables/test_configurable.py

```diff
@@ -83,3 +104,86 @@ def test_field_alias_set_configurable() -> None:
+    configurable_runnable = runnable.configurable_fields(
+        my_property=ConfigurableField(
+            id="my_property",
+            name="My property",
+            description="The property to test",
+        )
+    )
+    # first one
+    with pytest.raises(AttributeError):
+        configurable_runnable.not_my_custom_function()  # type: ignore[attr-defined]
+
+    assert configurable_runnable.my_custom_function() == "a"  # type: ignore[attr-defined]
+    assert (
+        configurable_runnable.my_custom_function_w_config(  # type: ignore[attr-defined]
+            {"configurable": {"my_property": "b"}}
+        )
+        == "b"
+    )
+    assert (
+        configurable_runnable.my_custom_function_w_config(  # type: ignore[attr-defined]
+            config={"configurable": {"my_property": "b"}}
+        )
+        == "b"
+    )
+
+    # second one
+    assert (
+        configurable_runnable.with_config(
+            configurable={"my_property": "b"}
+        ).my_custom_function()  # type: ignore[attr-defined]
+        == "b"
+    )
+
+
+def test_config_passthrough_nested() -> None:
+    runnable = MyRunnable(my_property="a")  # type: ignore
+    configurable_runnable = runnable.configurable_fields(
+        my_property=ConfigurableField(
+            id="my_property",
+            name="My property",
+            description="The property to test",
+        )
+    ).configurable_alternatives(
+        ConfigurableField(id="which", description="Which runnable to use"),
+        other=MyOtherRunnable(my_other_property="c"),
+    )
+    # first one
+    with pytest.raises(AttributeError):
+        configurable_runnable.not_my_custom_function()  # type: ignore[attr-defined]
+    assert configurable_runnable.my_custom_function() == "a"  # type: ignore[attr-defined]
+    assert (
+        configurable_runnable.my_custom_function_w_config(  # type: ignore[attr-defined]
+            {"configurable": {"my_property": "b"}}
+        )
+        == "b"
+    )
+    assert (
+        configurable_runnable.my_custom_function_w_config(  # type: ignore[attr-defined]
+            config={"configurable": {"my_property": "b"}}
+        )
+        == "b"
+    )
+    assert (
+        configurable_runnable.with_config(
+            configurable={"my_property": "b"}
+        ).my_custom_function()  # type: ignore[attr-defined]
+        == "b"
+    )
+    # second one
+    with pytest.raises(AttributeError):
+        configurable_runnable.my_other_custom_function()  # type: ignore[attr-defined]
+    with pytest.raises(AttributeError):
+        configurable_runnable.my_other_custom_function_w_config(  # type: ignore[attr-defined]
+            {"configurable": {"my_other_property": "b"}}
+        )
+    with pytest.raises(AttributeError):
+        configurable_runnable.with_config(
+            configurable={"my_other_property": "c", "which": "other"}
+        ).my_other_custom_function()  # type: ignore[attr-defined]
```

Comment:
should test test that my_other_custom_function does work if you configure to use alternative? would that work? (either way a test would help make expected behavior explicit)

Reply:
that's what this test is, see `which: other`. and no it doesn't work